File: ios/Classes/GalaceanPlayerViewFactory.swift
import Flutter
import UIKit

/**
 * Galacean Player View Factory
 * 用于创建 GalaceanPlayerView 实例
 */
class GalaceanPlayerViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    
    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }
    
    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        return GalaceanPlayerView(
            frame: frame,
            viewId: viewId,
            messenger: messenger,
            args: args
        )
    }
    
    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}

File: android/src/main/kotlin/github/kaierwen/galacean/galacean_native_player/GalaceanPlayerViewFactory.kt
package github.kaierwen.galacean.galacean_native_player

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

/**
 * Galacean Player View Factory
 * 用于创建 GalaceanPlayerView 实例
 */
class GalaceanPlayerViewFactory(
    private val messenger: BinaryMessenger
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return GalaceanPlayerView(context, viewId, messenger)
    }
}

